napalm_ios/ios.py

```diff
@@ -166,21 +171,30 @@ def _send_command(self, command):
             raise ConnectionClosedException(str(e))
 
     def is_alive(self):
-        """Returns a flag with the state of the SSH connection."""
+        """Returns a flag with the state of the connection."""
         null = chr(0)
-        try:
-            if self.device is None:
+        if self.device is None:
+            return {'is_alive': False}
+        if self.transport == 'telnet':
+            try:
+                # Try sending IAC + NOP
+                self.device.write_channel(telnetlib.IAC + telnetlib.NOP)
```

Comment:
That's interesting... Maybe worth adding a comment to explain (I had to look into to RFC to understand this).